### PR TITLE
Update to distribution/distribution 2.8.1

### PR DESCRIPTION
**Pull request by ph**
The changes is required because if you try download the dependencies it
will fails on the distributions module with a checksum error.

Recently they have moved the distributions package outside of the
docker organization and Dependabot detected a issues with the released
version.

How to reproduce the issue

 1. Take the master branch
 2. Clean your Golang module cache.
 3. GO111MODULE=on GOPROXY=direct go mod -v
 4. Apply this changes with remove the error.

Closes: #30716

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by ph** [comment]
I've tried to use replace rules to move from `docker/distribution` to `distribution/distribution`, I was not able to make `go mod tidy` not yield at me.

**Review by cmacknz — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -303,3 +303,6 @@ replace (
 	github.com/insomniacslk/dhcp => github.com/elastic/dhcp v0.0.0-20200227161230-57ec251c7eb3 // indirect
 	github.com/tonistiigi/fifo => github.com/containerd/fifo v0.0.0-20190816180239-bda0ff6ed73c
 )
+
+// Exclude this version because the version has an invalid checksum.
+exclude github.com/docker/distribution v2.8.0+incompatible
```
Comment (andrewkroh) [inline go.mod:308]: Is that needed because some packages has transitive dependencies on this 2.8.0?

Comment (ph) [reply go.mod:308]: For transitive dependencies, I wanted to guards against it.

**Comment by ymorin-orange** [comment]
@ph Thanks! :-)

**Pull request merged**